src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -0,0 +1,320 @@
+  {
+    integratePeak_(chromatogram, left, right);
+  }
+
+  void PeakIntegrator::integratePeak(const MSSpectrum& spectrum, const double& left, const double& right)
+  {
+    integratePeak_(spectrum, left, right);
+  }
+
+  double PeakIntegrator::simpson(MSChromatogram::ConstIterator it_begin, MSChromatogram::ConstIterator it_end) const
+  {
+    return simpson_(it_begin, it_end);
+  }
+
+  double PeakIntegrator::simpson(MSSpectrum::ConstIterator it_begin, MSSpectrum::ConstIterator it_end) const
+  {
+    return simpson_(it_begin, it_end);
+  }
+
+  void PeakIntegrator::calculatePeakShapeMetrics(const MSChromatogram& chromatogram, const double& left, const double& right)
+  {
+    calculatePeakShapeMetrics_(chromatogram, left, right);
+  }
+
+  void PeakIntegrator::calculatePeakShapeMetrics(const MSSpectrum& spectrum, const double& left, const double& right)
+  {
+    calculatePeakShapeMetrics_(spectrum, left, right);
+  }
+
+  double PeakIntegrator::getPeakArea() const { return peak_area_; }
+  double PeakIntegrator::getPeakHeight() const { return peak_height_; }
+  double PeakIntegrator::getPeakApexPos() const { return peak_apex_pos_; }
+  double PeakIntegrator::getBackgroundHeight() const { return background_height_; }
+  double PeakIntegrator::getBackgroundArea() const { return background_area_; }
+  PeakIntegrator::PeakShapeMetrics PeakIntegrator::getPeakShapeMetrics() const { return psm_; }
+
+  void PeakIntegrator::getDefaultParameters(Param& params)
+  {
+    params.clear();
+    params.setValue("integration_type", "intensity_sum", "The integration technique to use in integratePeak() and estimateBackground().");
+    params.setValidStrings("integration_type", ListUtils::create<String>("intensity_sum,simpson,trapezoid"));
+    params.setValue("baseline_type", "base_to_base", "The baseline type to use in estimateBackground().");
+    params.setValidStrings("baseline_type", ListUtils::create<String>("base_to_base,vertical_division"));
+  }
+
+  void PeakIntegrator::updateMembers_()
+  {
+    integration_type_ = (String)param_.getValue("integration_type");
+    baseline_type_ = (String)param_.getValue("baseline_type");
+  }
+
+  template <typename PeakContainerT>
+  void PeakIntegrator::estimateBackground_(const PeakContainerT& p, const double& left, const double& right)
+  {
+    const double int_l = p.PosBegin(left)->getIntensity();
+    const double int_r = (p.PosEnd(right)-1)->getIntensity();
+    const double delta_int = int_r - int_l;
+    const double delta_pos = (p.PosEnd(right)-1)->getPos() - p.PosBegin(left)->getPos();
+    const double min_int_pos = int_r <= int_l ? (p.PosEnd(right)-1)->getPos() : p.PosBegin(left)->getPos();
+    const double delta_int_apex = std::fabs(delta_int) * std::fabs(min_int_pos - peak_apex_pos_) / delta_pos;
+    background_height_ = std::min(int_r, int_l) + delta_int_apex;
+    double background = 0.0;
+    if (baseline_type_ == "base_to_base")
```

Comment:
minor comment: if you want to increase maintainability (and possible hard to find bugs) you could replace the string literals distributed all over the place by constants.
For example, if you define a: ```constexpr const char* INTEGRATION_TYPE_SIMPSON = "simpson"; ```
and use it throughout the code (and parameter), you don't run into the risk of hard to detect typos (aka using "simspon" at one position in the code).

Reply:
I think its fine, I would suggest however to throw an error if not found, e.g. 

```
if (baseline_type_ == "base_to_base")
{
}
else if (baseline_type_ == "vertical_division") 
{
}
else
  throw Exception
```